main.cpp

```diff
@@ -13,75 +13,584 @@ What you create in this project will be used as the basis of Project 5 in the co
+    3) Position Vertical
+    4) Rotation
+    5) Elevation
+3 things it can do:
+    1) send position
+    2) send value
+    3) send rotation
+ */
+
+struct FiducialMarker
+{
+    //1) Value (ID)
+    std::string markerID= "marker01";
+    //2) Position horizontal
+    int posX = 10;
+    //3) Position Vertical
+    int posY = 100;
+    //4) Rotation
+    int rotation = 0;
+    //5) Elevation
+    float elevation = 0.0f;
+
+    //1) send position
+    void sendMarkerPos(int X,int Y);
+    //2) send value
+    void sendMarkerID(std::string ID);
+    //3) send rotation
+    void sendMarkerRot(int rot);
+    //or really you just do all data in one call (avoids race condition)
+    void sendMarkerData(std::string ID, int X, int Y, float elev, int rot);
+};
+
+/*
+Thing 8) Audio Engine
+5 properties:
+    1) Master Volume
+    2) Section
+    3) Active instrument count
+    4) Active Effect count
+    5) External Input volume
+3 things it can do:
+    1) produce audio from instruments
+    2) apply effects to instruments
+    3) record sounds 
 */
+struct AudioEngine
+{
+    //1) Master Volume
+    float masterVol = 0.5f;
+    //2) Section
+    int Section = 3;
+    //3) Active instrument count
+    int activeInst = 4;
+    //4) Active Effect count
+    int activeEffects = 3;
+    //5) External Input volume
+    float extInputVol = 0.0f;
+    
+    //1) produce audio from instruments
+    void playAudio();
+    //2) apply effects to instruments
+    void applyEffect(int effect,int instNumber);
+    //3) record sounds
+    void recordSound(bool state);
+}; 
+
+/*
+Thing 9) MIDI Interface
+5 properties:
+    1) MIDI Channel
+    2) MIDI note number
+    3) MIDI velocity
+    4) MIDI Controller Value
+    5) MIDI Pitch Bend
+3 things it can do:
+    1) route MIDI data to audio engine
+    2) record MIDI input
+    3) play back MIDI recording
+ */
+struct MIDIinterface
+{
+    //1) MIDI Channel
+    int midiCh = 0;
+    //2) MIDI note number
+    int midiNote = 60;
+    //3) MIDI velocity
+    int midiVel = 80;
+    //4) MIDI Controller Value
+    int midiCtl = 7;
+    //5) MIDI Pitch Bend
+    double midiPitchBend = 134567;
+
+    //1) route MIDI data to audio engine
+    void routeMidi(int channel, int note, int vel);
+    //2) record MIDI input
+    void recordMidiIn(bool state);
+    //3) play back MIDI recording
+    void playMidiData(bool state);
+};
+/*
+Thing 10) Control Interface
+    1) Camera
+    2) Board
+    3) Fiducial Markers
+    4) Audio Engine
+    5) MIDI interface
+3 things it can do:
+    1) Detect position (horzontal and vertical) of hand or marker
+    2) Recognize different markers 
+    3) Send OSC messages
+ */
+struct ControlInterface
+{
+    //1) Camera
+    Camera camera;
+    //2) Board
+    Board board;
+    //3) Fiducial Markers
+    FiducialMarker marker;
+    //forgot to add hands so here they are
+    struct Hand
+    {
+      std::string name;
+      int posX;
+      int posY;
+      float elevation; 
+    };
```

Comment:
nested structs need 5 properties and 3 member functions
Fix this indentation
- [ ] revise